deployer/src/deployer/main.py

```diff
@@ -1,101 +1,99 @@
+    update_all_lambda_functions(dry_run=ctx.obj["dry_run"])
+
+
+@cli.command()
 @click.option(
     "--bucket",
-    default=DEFAULT_BUCKET,
-    help=f"Name of the bucket (default {DEFAULT_BUCKET!r})",
-)
-@click.option(
-    "--name",
-    default=DEFAULT_NAME,
-    help=f"Name of the site (default {DEFAULT_NAME_PATTERN!r})",
+    help='Name of the S3 bucket or one of "dev", "stage", or "prod"',
+    default=DEFAULT_BUCKET_NAME,
+    show_default=True,
 )
 @click.option(
-    "--bucket-location",
-    default=S3_DEFAULT_BUCKET_LOCATION,
-    help=(
-        f"Name of the S3 bucket location (like 'us-east-1') "
-        f"(default {S3_DEFAULT_BUCKET_LOCATION!r})"
-    ),
+    "--folder", help="Upload into this folder of the S3 bucket instead of its root",
 )
 @click.option(
-    "--refresh",  # XXX consider renaming this to "force-refresh"
+    "--force-refresh",
+    is_flag=True,
     default=False,
-    help="Ignores checking if files exist already",
     show_default=True,
-    is_flag=True,
+    help="Forces all files to be uploaded even if they already match what is in S3",
 )
 @click.option(
-    "--bucket-lifecycle-days",
-    required=False,
-    type=int,
-    help=(
-        "If specified, the number of days until uploaded objects are deleted. "
-        "(Only applicable when buckets are created!)"
-    ),
+    "--content-root",
+    help="The path to the root folder of the main content (defaults to CONTENT_ROOT)",
+    default=CONTENT_ROOT,
+    show_default=True,
 )
 @click.option(
-    "--dry-run",
-    default=False,
-    help="No actual uploading",
+    "--content-translated-root",
+    help="The path to the root folder of the translated content (defaults to CONTENT_TRANSLATED_ROOT)",
+    default=CONTENT_TRANSLATED_ROOT,
     show_default=True,
-    is_flag=True,
 )
 @click.option(
     "--no-progressbar",
+    help="Don't show the progress bar",
     default=DEFAULT_NO_PROGRESSBAR,
-    help="Don't use an iteractive progress bar",
     show_default=True,
     is_flag=True,
 )
 @click.argument("directory", type=click.Path())
+@click.pass_context
 def upload(ctx, directory, **kwargs):
-    p = Path(directory)
-    if not p.exists():
-        error(f"{directory} does not exist")
-        raise click.Abort
+    log.info(f"Deployer ({__version__})", bold=True)
+    content_roots = []
+    for name in ("content_root", "content_translated_root"):
+        value = kwargs.pop(name)
+        if not value:
+            log.warning(f"Warning: No {name.replace('_', '-')} has been specified.")
+            continue
+        content_root = Path(value)
+        if not content_root.is_dir():
+            raise click.ClickException(
+                f'The {name_hyphenated} "{content_root}" does not exist or is '
+                "not a directory."
+            )
+        content_roots.append(content_root)
+
+    dirpath = Path(directory)
+
+    if not dirpath.exists():
+        raise click.ClickException(f"{directory} does not exist")
```

Comment:
Thanks for the cool validation tips! I incorporated all of your suggestions except for:
```python
for fp in content_root.glob("**/_redirects.txt"):
```
I had originally used that, but it was significantly slower than
```python
            for fp in iterdir(content_root, max_depth=1):
                if fp.name != "_redirects.txt":
                    continue
```
Since the redirect files are always under the locale, I could use (note the one star):
```python
for fp in content_root.glob("*/_redirects.txt"):
```

Reply:
Slower, but can you even notice the difference?

Reply:
Oh. I didn't know `for fp in content_root.glob("*/_redirects.txt"):` was a thing even. But that actually makes sense. 
But this is all nits in a sense since your existing ad-hoc solution works perfectly fine.